jerry-core/parser/js/js-parser-statm.c

```diff
@@ -613,6 +634,15 @@ parser_parse_function_statement (parser_context_t *context_p) /**< context */
 
   JERRY_ASSERT (context_p->token.type == LEXER_KEYW_FUNCTION);
 
+#if ENABLED (JERRY_ES2015)
+  uint8_t flags = parser_statement_flags[context_p->stack_top_uint8];
+  if ((flags & PARSER_STATM_SINGLE_STATM)
+      && !(flags & PARSER_STATM_ALLOW_FUNC_DECL))
```

Comment:
This two check can be reduced to one with a PARSER_STATM_DISALLOW_FUNC_DECL flag.